core/src/main/scala/kafka/server/BrokerLifecycleManager.scala

```diff
@@ -292,7 +292,7 @@ class BrokerLifecycleManager(
     }
     val data = new BrokerRegistrationRequestData().
         setBrokerId(nodeId).
-        setIsMigratingZkBroker(isZkBroker).
+        setMigratingZkBrokerEpoch(zkBrokerEpochSupplier.map(_.apply()).getOrElse(-1)).
```

Comment:
again, I would really like to distinguish between the "not zk broker" and "zk broker doesn't have a broker epoch for some reason" cases

so if we are a zk broker but don't know the broker epoch for some reason we should throw an exception or something (or just retry)